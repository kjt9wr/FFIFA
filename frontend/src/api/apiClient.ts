import axios from "axios";

const apiClient = axios.create({
  baseURL: "http://localhost:5000",
  headers: {
    "Content-Type": "application/json",
  },
});

export default apiClient;

// apiClient.js
apiClient.interceptors.response.use(
  (response) => response,
  (error) => {
    console.error("API call failed:", error);
    // Handle specific error cases
    // if (error.response.status === 401) {
    //   // Unauthorized
    // } else if (error.response.status === 404) {
    //   // Not found
    // }
    return Promise.reject(error);
  }
);
