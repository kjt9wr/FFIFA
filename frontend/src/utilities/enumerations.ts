export enum KEEPER_CLASS_ENUM {
  KEEP = 1,
  FRANCHISE_TAG = 2,
  SUPERMAX = 3,
  ARBITRATION = 4,
}

export enum POSITION {
  QB = "QB",
  RB = "RB",
  WR = "WR",
  TE = "TE",
}

export const getKeeperClassText = (keeperClassEnum: number) => {
  switch (keeperClassEnum) {
    case 2:
      return "Franchise Tag";
    case 3:
      return "Supermax";
    case 4:
      return "Arbitration";
    default:
      return "Keep";
  }
};

export const ALERT_STATE = {
  NONE: "none",
  SUCCESS: "sucess",
  ERROR: "error",
};

export enum OWNER_ID_ENUM {
  KEVIN = "5e80d724b3bdaf3413316177",
  JUSTIN = "5e80d930b3bdaf3413316189",
  ALEX = "5e80dd6ab3bdaf34133161bd",
  LUIGI = "5e80da66b3bdaf341331619b",
  CHRISTIAN = "5e80e173b3bdaf3413316213",
  MATT = "5e80df96b3bdaf34133161ef",
  BRENT = "5e80db62b3bdaf34133161ab",
  MICHAEL = "5e80de37b3bdaf34133161cf",
  NIKOS = "5e80dedcb3bdaf34133161dd",
  CHINMAY = "5e80e07eb3bdaf3413316200",
  PATRICK = "5e80e1dab3bdaf3413316225",
  JEFF = "5e80e1deb3bdaf3413316226",
  CASEY = "66fb53a23cb8429bd448fd61",
}
