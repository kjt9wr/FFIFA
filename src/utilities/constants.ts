export const CURRENT_SEASON_YEAR = "2025";
export const SLEEPER_LEAGUE_ID = "1000436768526741504";
export const SLEEPER_LEAGUE_ID_2024 = "1124841056500727808";
export const DRAFT_ID = "1000436769222967296";
export const DRAFT_ID_2024 = "1124841056500727809";
