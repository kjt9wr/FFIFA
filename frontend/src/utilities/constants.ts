import { POSITION } from "./enumerations";

//configurable variable for the upcoming season
const UPCOMING_SEASON_YEAR = "2026";

export const RECORDED_YEARS = ["2020", "2021", "2022", "2023", "2024", "2025"];

export const BASE_CAP: Record<string, number> = {
  "2022": 281,
  "2023": 311,
  "2024": 338,
  "2025": 375,
  "2026": 416,
};

export const CURRENT_SLEEPER_LEAGUE_ID: Record<number, string> = {
  2023: "1000436768526741504",
  2024: "1124841056500727808",
  2025: "1257029927102849024",
};

const SLEEPER_DRAFT_ID: Record<number, string> = {
  2023: "1000436769222967296",
  2024: "1124841056500727809",
  2025: "1257029927123832832",
};

const getCurrentYear = () => {
  return Number(UPCOMING_SEASON_YEAR) - 1;
};

export const getUpcomingSeasonYear = () => UPCOMING_SEASON_YEAR;

export const getUpcomingYearIndex = () => {
  return Number(UPCOMING_SEASON_YEAR) - 2020;
};

export const getCurrentSleeperLeagueId = () => {
  const currentYear = getCurrentYear();
  return CURRENT_SLEEPER_LEAGUE_ID[currentYear];
};

export const getMostRecentDraftId = () => {
  const currentYear = getCurrentYear();
  return SLEEPER_DRAFT_ID[currentYear];
};

export const POSITIONS = [POSITION.QB, POSITION.RB, POSITION.WR, POSITION.TE];

export const BASE_URL_LOCALHOST = "http://localhost:5000/";
