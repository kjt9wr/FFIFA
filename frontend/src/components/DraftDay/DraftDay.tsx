import { Alert, Card, Col, Container, Row } from "reactstrap";
import { usePenaltyFees } from "../../custom-hooks/custom-hooks";
import { PenaltyFeeInfo } from "../../interfaces/interfaces";
import { calculateFollowingYearBaseCap } from "../../services/ffifa.service";
import { getUpcomingSeasonYear } from "../../utilities/constants";
import SpinnerWrapper from "../reusable/SpinnerWrapper";

const displayOwnersUnderTax = (penaltyFeeData: PenaltyFeeInfo[]) => {
  return penaltyFeeData
    .filter((owner: PenaltyFeeInfo) => owner.penaltyFee === 0)
    .map((currentOwner: PenaltyFeeInfo) => {
      return <div key={currentOwner.name}>{currentOwner.name}</div>;
    });
};

const displayOwnersInTax = (ownerData: PenaltyFeeInfo[]) => {
  return ownerData
    .filter((owner: PenaltyFeeInfo) => owner.penaltyFee > 0)
    .map((currentOwner: PenaltyFeeInfo) => {
      return (
        <div key={currentOwner.name}>
          {currentOwner.name} ${currentOwner.penaltyFee}
        </div>
      );
    });
};
/*
 * This component displays draft day information
 */
const DraftDay = () => {
  const { loading, error, penaltyFees, totalInPot, payoutPerOwner } =
    usePenaltyFees();

  const followingYear = Number(getUpcomingSeasonYear()) + 1;
  const followingYearCap = calculateFollowingYearBaseCap(penaltyFees);

  return (
    <Container>
      <h1 className="page-title"> Draft Day Info </h1>
      <SpinnerWrapper loading={loading} />
      {error && <Alert color="danger">Error fetching draft data</Alert>}
      {!error && !loading && (
        <>
          <h4>Total Money in the Pot: ${totalInPot}</h4>
          <h4>Payout Per Owner: ${payoutPerOwner}</h4>
          <h4>
            {followingYear} Base Cap: ${followingYearCap}
          </h4>
          <Card className="p-1 my-2">
            <Row>
              <Col>
                <h3>Under Tax: </h3>
                <h5>{displayOwnersUnderTax(penaltyFees)}</h5>
              </Col>
              <Col>
                <h3>Over Tax: </h3>
                <h5>{displayOwnersInTax(penaltyFees)}</h5>
              </Col>
            </Row>
          </Card>
        </>
      )}
    </Container>
  );
};

export default DraftDay;
