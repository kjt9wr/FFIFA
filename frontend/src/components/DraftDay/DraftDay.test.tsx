import { render, screen, waitFor } from "@testing-library/react";
import DraftDay from "./DraftDay";
import * as constants from "../../utilities/constants";
import apiClient from "../../api/apiClient";
import { OWNERS_FROM_DB } from "../__mockdata__/component.mockdata";
import * as ffifaService from "../../services/ffifa.service";

describe("draft day page", () => {
  beforeAll(() => {
    jest.spyOn(constants, "getUpcomingSeasonYear").mockReturnValue("2025");
    jest.spyOn(ffifaService, "calculateTotalInPot").mockReturnValue(100);
    jest.spyOn(ffifaService, "calculateLuxaryPotPayout").mockReturnValue(11);
    jest
      .spyOn(ffifaService, "calculateFollowingYearBaseCap")
      .mockReturnValue(416);
  });

  it("renders successfully", async () => {
    jest.spyOn(apiClient, "get").mockResolvedValue({ data: OWNERS_FROM_DB });
    const view = render(<DraftDay />);

    await waitFor(async () => {
      expect(await screen.findByText("Draft Day Info")).toBeTruthy();
    });

    expect(view).toMatchSnapshot();
  });

  it("displays an error upon fetching failure", async () => {
    jest.spyOn(apiClient, "get").mockRejectedValue({});

    render(<DraftDay />);

    expect(await screen.findByText("Error fetching draft data")).toBeTruthy();
  });
});
