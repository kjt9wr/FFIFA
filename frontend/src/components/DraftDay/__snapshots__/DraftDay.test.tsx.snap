// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`draft day page renders successfully 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <h1
          class="text-center"
        >
           Draft Day Info 
        </h1>
        <h4>
          Total Money in the Pot: $
          100
        </h4>
        <h4>
          Payout Per Owner: $
          11
        </h4>
        <h4>
          2026
           Base Cap: $
          416
        </h4>
        <div
          class="p-1 my-2 card"
        >
          <div
            class="row"
          >
            <div
              class="col"
            >
              <h3>
                Under Tax: 
              </h3>
              <h5>
                <div>
                  Christian
                </div>
                <div>
                  Brent
                </div>
                <div>
                  Matt
                </div>
                <div>
                  Casey
                </div>
              </h5>
            </div>
            <div
              class="col"
            >
              <h3>
                Over Tax: 
              </h3>
              <h5>
                <div>
                  Jeff
                   $
                  5
                </div>
                <div>
                  Kevin
                   $
                  34
                </div>
                <div>
                  Nikos
                   $
                  22
                </div>
                <div>
                  Justin
                   $
                  21
                </div>
                <div>
                  Luigi
                   $
                  64
                </div>
                <div>
                  Alex
                   $
                  47
                </div>
                <div>
                  Patrick
                   $
                  67
                </div>
                <div>
                  Michael
                   $
                  25
                </div>
              </h5>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <h1
        class="text-center"
      >
         Draft Day Info 
      </h1>
      <h4>
        Total Money in the Pot: $
        100
      </h4>
      <h4>
        Payout Per Owner: $
        11
      </h4>
      <h4>
        2026
         Base Cap: $
        416
      </h4>
      <div
        class="p-1 my-2 card"
      >
        <div
          class="row"
        >
          <div
            class="col"
          >
            <h3>
              Under Tax: 
            </h3>
            <h5>
              <div>
                Christian
              </div>
              <div>
                Brent
              </div>
              <div>
                Matt
              </div>
              <div>
                Casey
              </div>
            </h5>
          </div>
          <div
            class="col"
          >
            <h3>
              Over Tax: 
            </h3>
            <h5>
              <div>
                Jeff
                 $
                5
              </div>
              <div>
                Kevin
                 $
                34
              </div>
              <div>
                Nikos
                 $
                22
              </div>
              <div>
                Justin
                 $
                21
              </div>
              <div>
                Luigi
                 $
                64
              </div>
              <div>
                Alex
                 $
                47
              </div>
              <div>
                Patrick
                 $
                67
              </div>
              <div>
                Michael
                 $
                25
              </div>
            </h5>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
